migration/CMTOOL-284_JBoss_EAP_7.4_Server_Migration_Tool.adoc

```diff
@@ -0,0 +1,155 @@
+= JBoss EAP 7.4 Server Migration Tool
+:author:            Eduardo Martins
+:email:             emartins@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This document specifies the development of the JBoss EAP 7.4 server migration tool feature.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1516[EAP7-1516]
+
+=== Related Issues
+
+* https://issues.jboss.org/projects/CMTOOL/issues/CMTOOL-284[CMTOOL-284]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:okotek@redhat.com[Ondrej Kotek]
+
+=== Affected Projects or Components
+
+WildFly (WFLY Migration Component), JBoss Server Migration Tool (CMTOOL)
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* The following server migrations should be supported:
+** JBoss EAP 6.4 to JBoss EAP 7.4
+** JBoss EAP 7.3 to JBoss EAP 7.4
+
+* Migration from patched servers, i.e. EAP 6.4/7.4 CPs, should be supported.
+* Migration to patched servers, i.e. EAP 7.4 CPs, should be supported.
+* The server migration should include:
+** server configurations
+** modules
+** deployments
+
+==== Migration of Server Configurations
+
+* The migration of the following JBoss EAP 6.4 and 7.3 server configurations should be supported::
+** standalone server configurations
+** domain and host controller configurations
+
+* The migration of JBoss EAP 6.4 server configurations includes:
+** removal of configurations from subsystems not supported by JBoss EAP 7.4, and related extensions
+*** cmp
+*** configadmin
+*** jaxr
+*** osgi
+*** threads
+** migration of configurations from legacy subsystems deprecated by JBoss EAP 7.4, converting these to configurations of the JBoss EAP 7.3 subsystems providing similar functionality
+*** jacorb → iiop-openjdk
+*** messaging → messaging-activemq
+*** web → undertow
+** add default configurations for the following JBoss EAP 7.4 subsystems, which didn't exist in JBoss EAP 6.4, and are required for the server to work correctly
+*** core-management
+*** discovery
+*** ee-security
+*** elytron
+*** microprofile-config-smallrye
```

Comment:
The microprofile subsystems are no longer part of the base EAP.

There are two new subsystems in EAP 7.4: `health` and `metrics`.

Reply:
updated to remove these, and include the new health and metrics